src/App.tsx

```diff
@@ -17,18 +36,42 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                input={title}
+                setInput={setTitle}
+                selectedButton={filter}
+                setSelectedButton={setFilter}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todosLoading && (
+                <Loader />
+              )}
+              {!todosLoading && todos.length > 0 && (
+                <TodoList
+                  title={title}
+                  todos={todos}
+                  filter={filter}
+                  iconId={iconId}
+                  setIsModalOpen={setIsModalOpen}
+                  setTodoCard={setTodoCard}
+                  setIconId={setIconId}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {isModalOpen && (
```

Comment:
you could use `todoCard` instead of a special state variable `isModalOpen`. If `todoCard` is not `null` it means the user selected one and we should open the modal. Also, in this case, you would be sure that `todoCard` is always defined in the `TodoModal` component

Reply:
it could also help you to fix this test:
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/86923089/6fb233ea-f717-4524-984f-b6edb2fba2e2)